iroha_p2p/src/network.rs

```diff
@@ -278,13 +413,13 @@ pub type ConnectionId = u64;
 
 /// Variants of messages from [`Peer`] - connection state changes and data messages
 #[derive(Clone, Debug, iroha_actor::Message, Decode)]
-pub enum PeerMessage<T: Encode + Decode> {
+pub enum PeerMessage<T: Encode + Decode + Debug> {
     /// Peer just connected and finished handshake
-    Connected(PeerId, ConnectionId),
+    Connected(PeerId, ConnectionId, ConnectionId),
     /// Peer disconnected
-    Disconnected(PeerId),
+    Disconnected(PeerId, ConnectionId, ConnectionId),
```

Comment:
Do we need connection ids? We can identify connections by keypairs and keep the one which initiated peer with higher public key.

Reply:
Yes, we need connection ids.
When the peer is created it is saved in `new_peers` hashmap with the key == connection_id.
When the peer handshakes successfully with the other end we signal the `Network` that we are `Connected`, and it can get this `Peer` actor from `new_peers` and save into `peers` hashmap by the public key as key. Before that we don't know what public key will have that `Peer`.

Reply:
Yeah, that makes sence, but i guess it could be make without both:
- states (connected/disconnected)
- connection ids

Reply:
And how is that?